canvas/canvas_course_usage.py

```diff
@@ -0,0 +1,109 @@
+            return
+
+        try:
+            analytics_data = json.loads(response.result().text)
+        except JSONDecodeError as e:
+            logger.error(f"Error in parsing the response due to {e.msg}")
+            logger.info("Append to retry list")
+            self.course_retry_list.append(course_id)
+            return
+
+        if not analytics_data:
+            logger.info(f"Response for fetching canvas course usage is empty")
+            return
+        new_dic['course_id'] = course_id
+        new_dic['analytics'] = analytics_data
+
+        self.canvas_usage_courses.append(new_dic)
+
+    def _get_canvas_course_views_participation_data(self, retry_courses=None):
+        logger.info("Starting of _get_canvas_course_views_participation_data call")
+        with FuturesSession() as session:
+            headers = {'Content-type': 'application/json', 'Authorization': 'Bearer ' + self.canvas_token}
+            # https://umich.instructure.com/api/v1/courses/course_id/analytics/activity
+            if retry_courses is None:
+                logger.info("Initial round getting canvas_course_usage data")
+                responses = [session.get(f'{self.canvas_url}/api/v1/courses/{course_id}/analytics/activity',
+                                         headers=headers) for course_id in self.course_ids]
+            else:
+                self.course_retry_list = []
+                logger.info("Retry round getting canvas_course_usage data")
+                responses = [session.get(f'{self.canvas_url}/api/v1/courses/{retry_course}/analytics/activity',
+                                         headers=headers) for retry_course in retry_courses]
+
+            for response in as_completed(responses):
+                self.parsing_canvas_course_usage_data(response)
+
+        logger.info(f"Any thing to Retry? With List of length {len(self.course_retry_list)} : {self.course_retry_list}")
+        if len(self.course_retry_list) != 0 and self.retry_count < self.retry_attempts:
+            logger.info(f"Retrying again for {self.retry_count + 1}")
+            self.retry_count = self.retry_count + 1
+            self._get_canvas_course_views_participation_data(self.course_retry_list)
+
+    # preparing the data to be loaded to df in format [date, views, paticipations, course_id]
+    def canvas_course_usage_to_df(self):
+        rows = []
+        for data in self.canvas_usage_courses:
+            analytics = data['analytics']
+            course_id = data['course_id']
+
+            for row in analytics:
+                row['course_id'] = course_id
+                rows.append(row)
+
+        df = pd.DataFrame(rows)
+        logger.info(df.head())
+        df.drop(['id'], axis=1, inplace=True)
+        df_dup = df[df.duplicated()]
+        logger.info('Check for duplicate items ')
+        logger.info(df_dup)
+        df = df.drop_duplicates()
+        logger.info(df.head())
```

Comment:
@jonespm and I talked about the use of `inplace` a little bit, which I think will be deprecated in the future. See issue #52. Would you mind changing line 95 here to `df = df.drop(['id], axis=1)`. It's just a little more clear to me, and you're also not using `inplace` on line 99, so it'd be a little more consistent.